vault/resource_aws_secret_backend_static_role.go

```
@@ -0,0 +1,171 @@
+		},
+	}
+	return &schema.Resource{
+		CreateContext: createUpdateAWSStaticRoleResource,
+		UpdateContext: createUpdateAWSStaticRoleResource,
+		ReadContext:   ReadContextWrapper(readAWSStaticRoleResource),
+		DeleteContext: deleteAWSStaticRoleResource,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: fields,
+	}
+}
+
+var awsSecretBackendStaticRoleFields = []string{
+	consts.FieldName,
+	consts.FieldUsername,
+	consts.FieldRotationPeriod,
+}
+
+// createUpdateAWSStaticRoleResources upserts an aws static-role to our vault instance.
+func createUpdateAWSStaticRoleResource(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := provider.GetClient(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	backend := d.Get(consts.FieldBackend).(string)
+	role := d.Get(consts.FieldName).(string)
+	rolePath := fmt.Sprintf("%s/%s/%s", backend, awsStaticRolesAffix, role)
+	log.Printf("[DEBUG] Creating AWS static role at %q", rolePath)
+	data := map[string]interface{}{}
+	for _, field := range awsSecretBackendStaticRoleFields {
+		if v, ok := d.GetOk(field); ok {
+			data[field] = v
+		}
+	}
+
+	if _, err := client.Logical().WriteWithContext(ctx, rolePath, data); err != nil {
+		return diag.FromErr(fmt.Errorf("error writing %q: %s", rolePath, err))
+	}
+
+	d.SetId(rolePath)
+	log.Printf("[DEBUG] Wrote %q", rolePath)
+	return readAWSStaticRoleResource(ctx, d, meta)
+}
+
+func readAWSStaticRoleResource(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := provider.GetClient(d, meta)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	path := d.Id()
+	log.Printf("[DEBUG] Reading %q", path)
+	backend, role, err := parseAWSStaticRolePath(path)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("invalid id %q; %w", path, err))
+	}
+
+	err = d.Set(consts.FieldBackend, backend)
```

We should also return an error here if it is non-nil